components/SocketIOProvider.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { ReactNode, useEffect, useState } from "react"
+import { io, Socket } from "socket.io-client"
+
+export const SocketIOContext = React.createContext<Socket>(null as any)
+
+interface SocketIOProviderProps {
+  children: ReactNode
+}
+
+export default function SocketIOProvider({ children }: SocketIOProviderProps) {
```

Comment:
Rather than manage this in a provider, I would want to have the connection available in RxJS stream.  Then, we would use react-rxjs `bind` to hook up of state of components that need the stream.  

Conceptually, the model is similar to what we do here -
 https://github.com/AdaptiveConsulting/ReactiveTraderCloud/blob/master/src/new-client/src/services/client/endpoints.ts

though we're not using the Stomp protocol.  We'd likely use `fromEvent` from RxJS or handroll the Observable we need: 

https://stackoverflow.com/questions/29813503/how-to-use-rxjs-with-socket-io-on-event